### PR TITLE
Bluetooth: Mesh: Replace net_buf with mem_slabs

### PR DESCRIPTION
**Pull request by LingaoM**
Use memslabs for loopback, for save some ram footprint.

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

**Comment by LingaoM** [comment]
west build -b nrf52840dk_nrf52840 zephyr/samples/bluetooth/mesh

Before
1c85e0f5a6740474ca299e11db6d813



After

f3c4d51578f4a5e08012830556597f4

File: subsys/bluetooth/mesh/net.c [hunk 1]
```diff
@@ -98,8 +96,16 @@ struct bt_mesh_net bt_mesh = {
  */
 static bool ivi_was_recovered;
 
-NET_BUF_POOL_DEFINE(loopback_buf_pool, CONFIG_BT_MESH_LOOPBACK_BUFS,
-		    LOOPBACK_MAX_PDU_LEN, LOOPBACK_USER_DATA_SIZE, NULL);
+struct loopback_buf {
+	sys_snode_t node;
+	struct bt_mesh_subnet *sub;
+	uint8_t len;
+	uint8_t data[];
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/net.c:103]:
```suggestion
	uint8_t data[LOOPBACK_MAX_PDU_LEN];
```
to ensure that the slab definition and the buffer lengths are consistent

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
I only have a minor nitpick. Looks like a massive optimization, great stuff!

**Review by trond-snekvik — APPROVED** [review]

File: subsys/bluetooth/mesh/net.c [hunk 2]
```diff
@@ -98,8 +96,16 @@ struct bt_mesh_net bt_mesh = {
...
+struct loopback_buf {
+	sys_snode_t node;
+	struct bt_mesh_subnet *sub;
+	uint8_t len;
+	uint8_t data[LOOPBACK_MAX_PDU_LEN];
+};
+
+K_MEM_SLAB_DEFINE(loopback_buf_pool,
+		  sizeof(struct loopback_buf),
+		  CONFIG_BT_MESH_LOOPBACK_BUFS, __alignof__(struct loopback_buf));
```
Comment (alxelax) [inline subsys/bluetooth/mesh/net.c:108]: Is there any reason to align buffers to `struct loopback_buf`? Will it be enough to align to machine word size?

Comment (LingaoM) [reply subsys/bluetooth/mesh/net.c:108]: I think this method is more compatible.

Comment (alxelax) [reply subsys/bluetooth/mesh/net.c:108]: I tried both boundaries. Seems there is no difference in RAM size. I see it is used similarly in Bluetooth host but aligned onto hardcoded 4 bytes in drivers, fs and etc. No common usage way among components.
Then LGTM.

**Review by alxelax — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Comment by carlescufi** [comment]
@jhedberg waiting on an ack from you

**Review by jhedberg — APPROVED** [review]

**Pull request merged**